### PR TITLE
Add a basic components guide

### PR DESCRIPTION
**Pull request by pshrmn**
While the quick start guide provides a working example, I think that for some people it can be a little bit much to learn React Router by having to parse through a file to figure out what is going on.

This covers the basic components that should be known by a user in order to create a `react-router-dom` application. It could be adapted pretty easily for `react-router-native`, but that isn't my wheelhouse.

### CODE REVIEW

**Comment by timdorr** [comment]
Seems like a good idea to pair with a rapid start, copypasta kind of document like the Quick Start guide. 👍

File: packages/react-router-dom/docs/guides/basic-components.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+All of the components that you use in a web application should be imported from `react-router-dom`.
+
+```js
+import { BrowserRouter, Route, Link } from 'react-router-dom'
+```
+
+## Routers
+
+At the core of every React Router application should be a router component. For web projects, `react-router-dom` provides `<BrowserRouter>` and `<HashRouter>` routers. Both of these will create a specialized `history` object for you. Generally speaking, you should use a `<BrowserRouter>` if you have a server that responds to requests and a `<HashRouter>` if you are using a static file srever.
```
Comment (ojab) [inline packages/react-router-dom/docs/guides/basic-components.md]: typo `srever`

**Comment by pshrmn** [comment]
Huh, it has been six months since I opened this PR. I hadn't completely forgotten about it, but I had forgotten how long ago it was opened. I'm not sure if this will be merged, but I rebased it so at least it can be.

**Comment by timdorr** [comment]
I mean, I don't see any reason *not* to add it. In it goes...

**Pull request merged**